File: server/ruby/get-user-profile.rb
require 'dotenv'
require 'courier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
user_id = ENV['COURIER_GET_USER_PROFILE_USER_ID']

client = Courier::Client.new(api_key: api_key)

response = client.profiles.retrieve(user_id)

puts response

File: server/go/get-user-profile.go
//go:build ignore

// +build ignore

package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	"github.com/trycourier/courier-go/v4"
	"github.com/trycourier/courier-go/v4/option"
)

func LoadEnv() {
	envPath := filepath.Join("..", ".env")
	godotenv.Load(envPath)
}

func main() {
	LoadEnv()

	apiKey := os.Getenv("COURIER_API_KEY")
	userID := os.Getenv("COURIER_GET_USER_PROFILE_USER_ID")

	client := courier.NewClient(
		option.WithAPIKey(apiKey),
	)

	var response map[string]interface{}
	err := client.Get(
		context.Background(),
		fmt.Sprintf("/profiles/%s", userID),
		nil,
		&response,
	)

	if err != nil {
		panic(err.Error())
	}

	fmt.Printf("%+v\n", response)
}
